src/test/cpp/subsystems/intake_test.cpp

```diff
@@ -19,12 +21,131 @@ class IntakeTest : public testing::Test {
     IntakeSoftwareInterface sw_interface_;
 };
 
+/**
+ * Test to make sure the intake can be initialized, and test the intake config
+ */
 TEST_F(IntakeTest, InitIntake) {
     // Test intake initialization.
     ASSERT_TRUE(intake_->Init());
 
-    // Check to see that the interface updated
-    EXPECT_EQ(sw_interface_.reset_encoders, true);
+    // Check to see that the interface updated with initial configured values
     EXPECT_DOUBLE_EQ(sw_interface_.intake_config.open_loop_ramp_rate, 0.5);
+    EXPECT_DOUBLE_EQ(sw_interface_.intake_config.EXTENDED, 43.75);
+    EXPECT_DOUBLE_EQ(sw_interface_.intake_config.RETRACTED, 0);
+    EXPECT_DOUBLE_EQ(sw_interface_.intake_config.max_output_deploy, 0.8);
+    EXPECT_DOUBLE_EQ(sw_interface_.intake_config.max_output_retract, -0.8);
+    EXPECT_DOUBLE_EQ(sw_interface_.intake_config.max_indexer_current, 20);
+    EXPECT_EQ(sw_interface_.reset_encoders, true);
     EXPECT_EQ(sw_interface_.update_config, true);
 }
+
+/**
+ * Test to make sure the intake power (ie intake in cargo and/or spit it out) works
+ */
+TEST_F(IntakeTest, IntakePowerTest) {
+    const double kIntakePower = 1;
+    ASSERT_TRUE(intake_->SetIntakePower(kIntakePower));
+
+    EXPECT_EQ(sw_interface_.intake_power, kIntakePower);
+}
+
+/**
+ * Test to make sure the indexer power (ie intake in cargo and/or spit it out) works
+ */
+TEST_F(IntakeTest, IndexerPowerTest) {
+    const double kIndexerPower = -1;
+    ASSERT_TRUE(intake_->SetIndexerPower(kIndexerPower));
+
+    EXPECT_EQ(sw_interface_.indexer_power, kIndexerPower);
+}
+
+
+/**
+ * The big test.
+ * Tests that the intake position (deploy/extend) logic works
+ */
+TEST_F(IntakeTest, IntakePositionTest) {
```

Comment:
This is a fine test, but in the future it could be better to test the extend and retract separately.

better yet would be to create a helper function called TestIntake() that you can throw in a for loop and give initial conditions and expected values such as:
```
for (int i = 0; i < num_tests; ++i) {
    EXPECT_TRUE(TestIntake(inputs, expected_values));
}
```
but for now this is fine